lib/connect.js

```diff
@@ -605,7 +604,10 @@ function normalizeVideoCodecSettings(nameOrSettings) {
 }
 
 function validateBandwidthProfile(bandwidthProfile) {
-  if (bandwidthProfile === null || !isNonArrayObject(bandwidthProfile)) {
+  if (typeof bandwidthProfile === 'undefined' || bandwidthProfile === null) {
```

Comment:
I am debating if  we should allow explicit `null` and `undefined` as valid values. Not specifying bandwidthProfile is one thing, but if specified maybe we should require valid object.

Do we have other properites that allow  `{}` as well as `undefined` but they behave differently ?

Reply:
@makarandp0 I think not specifying the property, setting it to null or undefined is essentially the same thing. We do have `EncodingParameters` which is passed to `LocalParticipant.setParameters` where passing an `{}` does not change existing behavior.

Reply:
@manjeshbhargav  in this case  caller would get different behavior when they pass  `{}` (default bwprofile) v/s `null` or `undefined` (no bandwidth profile). I do not have any concerns with `{}` - but just wondering what is the reason to accept `null` and `undefined` values.

Reply:
@makarandp0 The `undefined` check is not for explicitly set `undefined` value, since we filter those props out. It is to check whether the property exists. I will change the code to reject `null` in order to remove any confusion.